### ISSUE TITLE
Dropout layer on sparse_tensor caculate?

### ISSUE THREAD

**Issue by yanty123** [pytorch/pytorch#35798]
## ❓ Questions and Help
I‘d like to know if there is a dropout layer that i can use it on sparse tensor.
I will be appreciate it if you can show me the name? Just like the sparse_dropout in tensorflow!

We have a set of [listed resources available on the website](https://pytorch.org/resources). Our primary means of support is our discussion forum:

- [Discussion Forum](https://discuss.pytorch.org/)

**Comment by ezyang** [comment]
We don't currently have dropout on sparse tensors implemented, sorry!

**Comment by yanty123** [comment]
> We don't currently have dropout on sparse tensors implemented, sorry!

okay,thank you ! Maybe I have to consider to do it.

**Comment by johnbowen42** [comment]
Hi, is anyone working on this right now?  I'd like to try and work on it.

**Comment by gau-nernst** [comment]
I also wish to see Dropout supporting Sparse tensor. It is quite popular for graph neural networks. Meanwhile, I use the following hack (for COO sparse tensor)

```python
def sparse_dropout(x: torch.Tensor, p: float, training: bool):
    x = x.coalesce()
    return torch.sparse_coo_matrix(x.indices(), F.dropout(x.values(), p=p, training=training), size=x.size())
```
